### PR TITLE
BUGZ-21, Case 22219: Fix skyboxes/ambient maps on AMD

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.atlassian.net/browse/BUGZ-21
https://highfidelity.manuscript.com/f/cases/22219/Improve-Ambient-Lighting-with-better-ambient-map-HDR-and-Convolution

- Rebakes the default skybox
- Adds a versioning system to our .texmeta.json format
- Minor cleanup
- FrameReader now supports loading resources that start with ":"
- Fix the skybox issue

### CODE REVIEW

File: libraries/graphics/src/graphics/skybox.slf [hunk 1]
```diff
@@ -26,13 +26,6 @@ layout(location=0) in vec3  _normal;
...
-    vec3 color = skybox.color.rgb;
-
-    // blend is only set if there is a cubemap
-    float check = float(skybox.color.a > 0.0);
-    color = mix(color, texture(cubeMap, coord).rgb, check);
-    color *= mix(vec3(1.0), skybox.color.rgb, check * float(skybox.color.a < 1.0));
-
-    _fragColor = vec4(color, 0.0);
+    vec3 skyboxColor = texture(cubeMap, normalize(_normal)).rgb;
+    _fragColor = vec4(mix(skybox.color.rgb, skyboxColor, skybox.color.a), 1.0);
```
Comment (SamGondelman) [inline libraries/graphics/src/graphics/skybox.slf]: not the fix but the old logic was unnecessary, skybox.color.a is set in Skybox::updateSchemaBuffer()

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19577/HighFidelity-Beta-PR15602-b92e267.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19577/HighFidelity-Quest-Beta-PR15602-b92e267.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19577/HighFidelity-Beta-PR15602-b92e267.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19577/HighFidelity-Beta-Interface-PR15602-b92e267.exe

File: libraries/gpu/src/gpu/FrameReader.cpp [hunk 2]
```diff
@@ -324,6 +323,13 @@ TexturePointer Deserializer::readTexture(const json& node, uint32_t external) {
     readOptional(ktxFile, node, keys::ktxFile);
     Element ktxTexelFormat, ktxMipFormat;
     if (!ktxFile.empty()) {
+        // If we get a texture that starts with ":" we need to re-route it to the resources directory
+        if (ktxFile.at(0) == ':') {
+            QString frameReaderPath = __FILE__;
+            frameReaderPath.replace("\\", "/");
+            frameReaderPath.replace("libraries/gpu/src/gpu/framereader.cpp", "interface/resources", Qt::CaseInsensitive);
+            ktxFile.replace(0, 1, frameReaderPath.toStdString());
+        }
```
Comment (SamGondelman) [inline libraries/gpu/src/gpu/FrameReader.cpp]: Is there a cleaner way to do this?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19643/HighFidelity-Beta-PR15602-dc3b2d9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19643/HighFidelity-Beta-Interface-PR15602-dc3b2d9.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19643/HighFidelity-Beta-PR15602-dc3b2d9.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19643/HighFidelity-Quest-Beta-PR15602-dc3b2d9.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19643/HighFidelity-Beta-PR15602-dc3b2d9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19643/HighFidelity-Beta-Interface-PR15602-dc3b2d9.exe

File: libraries/gpu/src/gpu/FrameReader.cpp (hunk 2 above)
Comment (jherico) [reply libraries/gpu/src/gpu/FrameReader.cpp]: We could have the frame player load up the binary resources.rcc file, so the requests for `:/` resources would succeed.  Long term we should better organize our resources so it's not all one giant .rcc file

**Pull request merged**